rickroll.py

```diff
@@ -153,5 +174,55 @@ def original():
 
     return str(play_tune(tune))
 
+@app.route("/sms", methods=["POST"])
+def sms():
+    """Adds an incoming SMS to a queue, to reply to later."""
+    bucket.put_object(
+        Key='queue/{to}/{from_}'.format(to=request.form['To'],
+                                        from_=request.form['From']),
+        Body='',
+    )
+    return "Hello world!"
+
+def send_sms():
+    """Empties the queue of incoming SMSes, replying to each one."""
+    for queue_entry in bucket.objects.filter(Prefix='queue/'):
+        _, our_number, their_number = queue_entry.key.split("/")
+        state_key = "state/{}/{}".format(our_number, their_number)
+
+        # Find out which message we sent last, if any
+        # Error handling taken from https://stackoverflow.com/a/33843019
+        try:
+            state_obj = s3.Object(
+                os.environ['data_bucket'],
+                state_key,
+            )
+            state_obj.load()
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == "404":
+                last_msg = -1
+            else:
+                raise
+        else:
```

Comment:
Yep!

You can also attach an `else` block to a `for` or `while` loop, which executes if the loop runs to completion (as opposed to hitting a `break`).